depth-test_z-acc_spatial-noise/depth_test.py

```diff
@@ -0,0 +1,174 @@
+import open3d as o3d
+import numpy as np
+from utils import *
+
+class DepthTest:
+	def __init__(self):
+		self.camera_dir = np.array([0, 0, -1])
+		self.plane_normal = np.array([0, 0, -1])
+		self.camera_wall_distance = 1 # m
+		self.plane_distance = -self.camera_wall_distance
+		self.plane_coeffs = (0,0,-self.camera_wall_distance)
+
+		self.point_cloud_corrected = o3d.geometry.PointCloud()
+		self.plane_fit_corrected_pcl = o3d.geometry.PointCloud()
+		self.plane_fit_pcl = o3d.geometry.PointCloud()
+		self.plane_fit_visualization = False
+
+		self.fitted = False
+
+		self.z_accuracy_medians = []
+		self.z_means = []
+		self.spatial_noise_rmses = []
+		self.samples = 0
+
+	def set_ground_truth(self, point_cloud: o3d.geometry.PointCloud):
+		points = np.asarray(point_cloud.points)
+		G = np.ones_like(points)
+		G[:, 0] = points[:, 0]
+		G[:, 1] = points[:, 1]
+		Z = points[:, 2]
+
+		self.camera_wall_distance = -np.mean(Z)
+		return self.camera_wall_distance
+
+	def fit_plane(self, point_cloud: o3d.geometry.PointCloud):
+		points = np.asarray(point_cloud.points)
+		G = np.ones_like(points)
+		G[:, 0] = points[:, 0]
+		G[:, 1] = points[:, 1]
+		Z = points[:, 2]
+		(k_x, k_y, k_z), resid, rank, s = np.linalg.lstsq(G, Z, rcond=None)
+		normal = np.array([k_x, k_y, -1])
+		normal = normal / np.linalg.norm(normal)
+
+		self.plane_normal = normal
+		self.plane_distance = k_z
+		self.plane_coeffs = (k_x, k_y, k_z)
+
+		self.fitted = True
+
+		return normal, k_z
```

Comment:
This fit is happening on OAK's PointCloud or Astra ?

Reply:
Depends on which camera is selected when you press the `f` key.

If we can ensure the cameras are perfectly parallel it wuld be preferable to use Astra for this. Otherwise I think it would be better to fit a plane on the OAK's point cloud to avoid possible missalignment errors.